scripts/update_jobs.py

```
@@ -656,15 +682,19 @@ def fetch_greenhouse_jobs(company_name: str, url: str, max_retries: int = 2, tim
-            else:
-                print(f"  ❌ Request error for {company_name} after {max_retries + 1} attempts: {e}")
+
+        except requests.exceptions.HTTPError as e:
+            if hasattr(e, 'response') and e.response is not None:
+                if not is_retryable_status(e.response.status_code):
+                    print(f"  ⚠️  {company_name}: HTTP {e.response.status_code} (non-retryable)")
+                    break
+                if attempt < max_retries:
+                    print(f"  ⚠️  Request error for {company_name}: {e}, retrying...")
+                    continue
+                else:
```

Ruff (0.15.5)

[warning] Unnecessary `else` after `continue` statement

Remove unnecessary `else`

(RET507)